Kdesktop 689 alllow kDrive client to start kDrive server

herve-er opened the pull request.

luc-guyot-infomaniak commented on src/gui/appclient.cpp:
```diff
@@ -645,6 +628,89 @@ void AppClient::askUserToLoginAgain(int userDbId, QString userEmail, bool invali
+		QMessageBox msgBox;
+		msgBox.setText(tr("No comm port provided! Clean restart in progress..."));
+		msgBox.exec();
+        startServer(true /*Close the client and let the server restart it*/);
```
Restart the server, close the client and let the server restart the client?

ClementKunz commented on src/gui/appclient.cpp:
```diff
@@ -645,6 +628,90 @@ void AppClient::askUserToLoginAgain(int userDbId, QString userEmail, bool invali
+        QMessageBox msgBox;
+        msgBox.setText(tr("Unable to connect to application server!"));
+        msgBox.exec();
+        //startServer(true /*Close the client and let the server restart it*/);
```
I don't think so, for me this is a totally different issue.
Herve's fix aims to automatically restart the app in case on crash.
This part of code handle the case where the server takes too much time to reply. Not ideal perhaps but unrelated to this PR. What do you think?

ClementKunz commented on src/gui/appclient.cpp:
```diff
@@ -645,6 +628,90 @@ void AppClient::askUserToLoginAgain(int userDbId, QString userEmail, bool invali
+        QMessageBox msgBox;
+        msgBox.setText(tr("The application server did not respond on time!"));
+        msgBox.exec();
+        //startServer(true /*Close the client and let the server restart it*/);
```
Same as above

ClementKunz commented on src/server/appserver.h:
```diff
@@ -121,6 +121,7 @@ class AppServer : public SharedTools::QtSingleApplication {
         bool _settingsAsked = false;
         bool _synthesisAsked = false;
         bool _clearKeychainKeysAsked = false;
+        bool _noStartClient = false;
```
Agreed, always try to avoid double negation

herve-er commented on src/server/appserver.cpp:
```diff
@@ -105,7 +122,8 @@ AppServer::AppServer(int &argc, char **argv)
       _helpAsked(false),
       _versionAsked(false),
       _clearSyncNodesAsked(false),
-      _debugMode(false) {
```
I left it as it was previously, but yes, it could be beneficial. Is there a technical advantage to initialize these members it in the header? For better readability?

herve-er commented:
We decided not to pursue the option of restarting the server without also restarting the client in the event of a server crash. This would have offered a marginally improved user experience by avoiding the need to close the UI. However, it could have led to numerous potential bugs, such as users interacting with the kDrive_client when the kDrive server is down (timeout, out-of-sync cache, etc.). While manageable, addressing these issues would be time-consuming. This matter will be discussed during the next overhaul of the UI.
